### PR TITLE
Add auto extract timestamp parameter support

### PR DESCRIPTION
**Pull request by ludovic-boutros**
We need to be able to enable the ["auto extract timestamp"](https://docs.splunk.com/Documentation/Splunk/9.1.1/RESTREF/RESTinput#services.2Fcollector.2Fevent) feature for a project.

This parameter is not exposed yet and therefore cannot be set on the HEC URI.

This PR expose this parameter.
It is not set by default to be keep compatibility with current behavior.

I have created a specific HEC URI builder in order to ease testing.

### CODE REVIEW

File: src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java [hunk 1]
```diff
@@ -187,6 +189,7 @@ public final class SplunkSinkConnectorConfig extends AbstractConfig {
     static final String SSL_TRUSTSTORE_PATH_DOC = "Path on the local disk to the certificate trust store.";
     static final String SSL_TRUSTSTORE_TYPE_DOC = "Type of the trust store (JKS, PKCS12, ...).";
     static final String SSL_TRUSTSTORE_PASSWORD_DOC = "Password for the trust store.";
+    static final String AUTO_EXTRACT_TIMESTAMP_DOC = "Sends timestamped events to HTTP Event Collector using the Splunk platform JSON event protocol when auto_extract_timestamp is set to \"true\" in the /event URL.";
```
Comment (VihasMakwana) [inline src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java]:
```suggestion
    static final String AUTO_EXTRACT_TIMESTAMP_DOC = "When set to true, it forces Splunk HEC to extract the timestamp from event envelope/event data";
```

File: README.md [hunk 2]
```diff
@@ -196,10 +197,11 @@ Use the below schema to configure Splunk Connect for Kafka
...
+|--------           |----------------------------|---------------|
+| `splunk.hec.json.event.enrichment` |  Only applicable to /event HEC endpoint. This setting is used to enrich raw data with extra metadata fields. It contains a list of key value pairs separated by ",". The configured enrichment metadata will be indexed along with raw event data by Splunk software. </br> **NOTE:** <br/> Data enrichment for /event HEC endpoint is only available in Splunk Enterprise 6.5 and above. By default, this setting is empty. See ([Documentation](http://dev.splunk.com/view/event-collector/SP-CAAAE8Y#indexedfield)) for more information. <br/>**Example:** `org=fin,bu=south-east-us`|               |
+| `splunk.hec.track.data` |  When set to `true`, data loss and data injection latency metadata will be indexed along with raw data. This setting only works in conjunction with /event HEC endpoint (`"splunk.hec.raw" : "false"`). Valid settings are `true` or `false`. | `false`       |
+| `splunk.hec.auto.extract.timestamp` |  Sends timestamped events to HTTP Event Collector using the Splunk platform JSON event protocol when auto_extract_timestamp is set to `true` in the /event URL. | `unset`       |
```
Comment (VihasMakwana) [inline README.md]:
```suggestion
| `splunk.hec.auto.extract.timestamp` | When set to `true`, it forces Splunk HEC to extract the timestamp from the event envelope/event data. See [/services/collector/event](https://docs.splunk.com/Documentation/Splunk/9.1.1/RESTREF/RESTinput#services.2Fcollector.2Fevent) for more details | `unset`       |
```

**Review by VihasMakwana — CHANGES_REQUESTED** [review]

**Comment by VihasMakwana** [comment]
Looks good other than a couple of nits. Also, my CI has been failing lately so I'll merge it only after I get it green.
It might take some time. I'll approve the PR though.

Thanks for the change, @ludovic-boutros. Appreciate it!

**Comment by ludovic-boutros** [comment]
@VihasMakwana It should be good now 👍 
Thank you.

**Review by VihasMakwana — APPROVED** [review]

**Comment by ludovic-boutros** [comment]
Thank you @VihasMakwana. 
Any idea of next release date ?

**Comment by VihasMakwana** [comment]
@ludovic-boutros Released in 2.2.0, let me know if it's good for you.

**Comment by ludovic-boutros** [comment]
That's perfect, thank you 👍

**Pull request merged**